operator/v1/types_console.go

```diff
@@ -31,6 +31,27 @@ type ConsoleSpec struct {
 	Customization ConsoleCustomization `json:"customization"`
 	// providers contains configuration for using specific service providers.
 	Providers ConsoleProviders `json:"providers"`
+	// route contains hostname and secret reference that contains the serving certificate for replacing the console's default route.
+	// If a custom route is specified, a new route will be created with the provided hostname.
+	// The default console route will be maintained to reserve the default hostname for console if the custom route is removed.
+	// If not specified, default route will be used.
+	// +optional
+	Route *ConsoleConfigRoute `json:"route,omitempty"`
+}
+
+// ConsoleConfigRoute holds information on external route access to console.
+type ConsoleConfigRoute struct {
+	// hostname is the desired custom address under which console will be available.
+	Hostname string `json:"hostname,omitempty"`
```

Comment:
If this hostname is not a subdomain of the default ingress domain, seems like the admin would have to manually configure DNS to point at the ingress controller. Maybe I'm misunderstanding something.

Reply:
Just update the PR adding notification regarding that case